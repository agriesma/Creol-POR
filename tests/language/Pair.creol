class Test
begin
  op init == var x: [Int, Int]; x := (1, 2)
end
