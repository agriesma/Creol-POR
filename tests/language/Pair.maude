load creol-interpreter
mod PROGRAM is
protecting CREOL-SIMULATOR .
op classes : -> Configuration [ctor] .
eq classes =
< "Test" : Cl | Inh: noInh, Par: noVid, Att: "x" |-> null, Mtds: 
  < "run" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return ( emp ) > *
  < "init" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: "x" ::= pair(int(1) , int(2)) ; return ( emp ) >, Ocnt: 0 >

.
endm
