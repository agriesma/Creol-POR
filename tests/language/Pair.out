==========================================
reduce in PROGRAM : classes .
result Class: 
< "Test" : Cl | Inh: noInh, Par: noVid, Att: "x" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: "x" ::= pair(
    int(1), int(2)) ; return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return(
    emp) >,
  Ocnt: 0 >

==========================================
rewrite in PROGRAM : classes main("Test", emp) .
result Configuration: 
< "Test" : Cl | Inh: noInh, Par: noVid, Att: "x" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: "x" ::= pair(
    int(1), int(2)) ; return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return(
    emp) >,
  Ocnt: 1 >
 
< ob("Test0") : "Test" |
  Att: "this" |-> ob("Test0"), "x" |-> pair(int(1), int(2)),
  Pr: idle,
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: noMsg,
  Lcnt: 2 >
 
< ob("main") : "" |
  Att: noSubst,
  Pr: idle,
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: noMsg,
  Lcnt: 0 >

Bye.
